### PR TITLE
New Expanse Content Pack

### PR DESCRIPTION
**Pull request by fvigo**
**READY FOR REVIEW**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
Expanse Pack for Cortex XSOAR

## Minimum version of Demisto
- [ ] 5.0.0
- [ ] 5.5.0
- [x] 6.0.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [ ] No

## Must have
- [x] Tests
- [x] Documentation 

Co-Author: @jtschichold

### CODE REVIEW

**Comment by fvigo** [comment]
** DO NOT REVIEW THIS NOW, BUT KEEP PR OPEN **

**Comment by fvigo** [comment]
Now ready for review, only some documentation is missing.

File: Packs/ExpanseV2/Scripts/PrismaCloudAttribution/PrismaCloudAttribution.py [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    return out_dict
+
+
+''' COMMAND FUNCTION '''
+
+
+def attribution_command(args: Dict[str, Any]) -> CommandResults:
+
+    assets = argToList(args.get('assets', []))
+    # demisto.debug(f"Assets: {assets!r}")
```
Comment (Bargenish) [inline Packs/ExpanseV2/Scripts/PrismaCloudAttribution/PrismaCloudAttribution.py:51]:
```suggestion
```

File: Packs/ExpanseV2/Scripts/PrismaCloudAttribution/PrismaCloudAttribution.py [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    fields = argToList(
+        args.get('fields', 'id,cloudType,resourceName,resourceType,regionId,accountId,accountName,hasAlert,service,ip,fqdn'))
+    # demisto.debug(f"Fields: {fields!r}")
```
Comment (Bargenish) [inline Packs/ExpanseV2/Scripts/PrismaCloudAttribution/PrismaCloudAttribution.py:54]:
```suggestion
```

File: Packs/ExpanseV2/Scripts/PrismaCloudAttribution/PrismaCloudAttribution.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+args:
+- description: List of Prisma Cloud assets to return.
+  isArray: true
+  name: assets
+- defaultValue: id,cloudType,resourceName,resourceType,regionId,accountId,accountName,hasAlert,service,ip,fqdn
+  description: Fields to be returned (comma separated string).
+  name: fields
```
Comment (Bargenish) [inline Packs/ExpanseV2/Scripts/PrismaCloudAttribution/PrismaCloudAttribution.yml:7]: Should have the `isArray` field set to true

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py [hunk 4]
```diff
@@ -0,0 +1,747 @@
...
+            if pagination is None:
+                break
+            next_url = pagination.get('next', None)
+            if next_url is None:
+                break
+
+            params = None
+
+    def get_iprange_by_id(self, iprange_id: str) -> Dict[str, Any]:
+        result: Dict = self._http_request(
```
Comment (Bargenish) [inline Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py:111]:
```suggestion
        result = self._http_request(
```

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py [hunk 5]
```diff
@@ -0,0 +1,747 @@
...
+
+def ip_to_demisto_indicator(ip_indicator: Dict[str, Any]) -> Optional[Dict[str, Any]]:
+    value = ip_indicator.get('ip', None)
+    if value is None:
+        return None
+
+    provider_name: Optional[str] = None
+    provider = ip_indicator.get('provider', None)
+    if provider is not None:
+        provider_name = provider.get('name', None)
```
Comment (Bargenish) [inline Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py:272]: suggestion - you can do this instead:
```suggestion
    provider = ip_indicator.get('provider', {})
    provider_name = provider.get('name', None)
```
*same for other occurances*

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py [hunk 6]
```diff
@@ -0,0 +1,747 @@
...
+    tenant = domain_indicator.get('tenant', None)
+    if tenant is not None:
+        tenant_name = tenant.get('name', None)
+
+    business_unit_names: List[str] = []
+    business_units = domain_indicator.get("businessUnits", [])
+    for bu in business_units:
+        if 'name' not in bu:
+            continue
+        business_unit_names.append(bu['name'])
```
Comment (Bargenish) [inline Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py:428]:
```suggestion
    business_units = domain_indicator.get("businessUnits", [])
    business_unit_names = [bu['name'] for bu in business_units if bu.get('name')]

```

File: Packs/ExpanseV2/Integrations/ExpanseV2/ExpanseV2.py [hunk 7]
```diff
@@ -0,0 +1,2330 @@
...
+from collections import defaultdict
+import ipaddress
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+
+""" CONSTANTS """
+
+SERVER = "https://expander.expanse.co"
```
Comment (Bargenish) [inline Packs/ExpanseV2/Integrations/ExpanseV2/ExpanseV2.py:32]: not being used

File: Packs/ExpanseV2/Integrations/ExpanseV2/ExpanseV2.py [hunk 8]
```diff
@@ -0,0 +1,2330 @@
...
+TOKEN_DURATION = 7200
+DEFAULT_RESULTS = 20  # default results per search
+MAX_RESULTS = 5000  # max results per search
+MAX_PAGE_SIZE = 1000  # max results per page
+MAX_INCIDENTS = 100  # max incidents per fetch
+MAX_UPDATES = 100  # max updates received
+PREFIX = SERVER + "/api"
```
Comment (Bargenish) [inline Packs/ExpanseV2/Integrations/ExpanseV2/ExpanseV2.py:39]: not being used

File: Packs/ExpanseV2/Integrations/ExpanseV2/ExpanseV2.py [hunk 9]
```diff
@@ -0,0 +1,2330 @@
...
+DATE_FORMAT = "%Y-%m-%dT%H:%M:%S.%fZ"
+PAGE_SIZE = 200
```
Comment (Bargenish) [inline Packs/ExpanseV2/Integrations/ExpanseV2/ExpanseV2.py:41]: not being used

File: Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.yml [hunk 10]
```diff
@@ -0,0 +1,56 @@
...
+  description: Comma separate list of field names to be used as vsys.
+  defaultValue: vsys
+- name: sightings_fields
+  description: Comma separated list of field names to be considered sighting counts.
+  defaultValue: count
+- name: source_ip_fields
+  description: Comma separated list of field names to be considered as source IPs.
+  defaultValue: src,src_ip
+- name: internal_ip_networks
+  description: Comma separated list of IPv4 Networks to be considered internal (default to RFC private networks).
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.yml:28]: @Bargenish  The value should be a comma separated list of networks as I am using argToList inside the code to parse this argument. I don't think this argument should be marked as an array, but as a single string.
Same as here: https://github.com/demisto/content/blob/master/Packs/FeedMitreAttack/Integrations/FeedMitreAttack/FeedMitreAttack.yml#L98 paired with the code: https://github.com/demisto/content/blob/master/Packs/FeedMitreAttack/Integrations/FeedMitreAttack/FeedMitreAttack.py#L454
There are multiple occurences of this pattern in different integrations.

File: Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.py [hunk 11]
```diff
@@ -0,0 +1,128 @@
...
+
+from typing import Dict, List, Any, Tuple
+from ipaddress import IPv4Address, IPv4Network
+import traceback
+
+
+''' STANDALONE FUNCTION '''
+
+
+def is_internal(net_list: List[IPv4Network], ip: IPv4Address) -> bool:
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.py:17]: Added

File: Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.py [hunk 12]
```diff
@@ -0,0 +1,128 @@
...
+    if net_list is None or len(net_list) == 0:
+        return ip.is_private
+
+    result = next((inetwork for inetwork in net_list if ip in inetwork), None)
+    return result is not None
+
+
+def deconstruct_entry(entry: Dict[str, str],
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.py:25]: Added

File: Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.py [hunk 13]
```diff
@@ -0,0 +1,128 @@
...
+                      serial_fields: List[str],
+                      vsys_fields: List[str],
+                      sightings_fields: List[str],
+                      source_ip_fields: List[str]) -> Tuple[Optional[str],
+                                                            Optional[str],
+                                                            Optional[str],
+                                                            Optional[int]]:
+    serial = next((entry[field] for field in serial_fields if field in entry), None)
+    vsys = next((entry[field] for field in vsys_fields if field in entry), None)
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.py]: Applied

File: Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionIP/ExpanseAggregateAttributionIP.py [hunk 14]
```diff
@@ -0,0 +1,106 @@
...
+
+from typing import Dict, List, Any, Tuple
+from ipaddress import IPv4Address, IPv4Network
+import traceback
+
+
+''' STANDALONE FUNCTION '''
+
+
+def is_internal(net_list: List[IPv4Network], ip: IPv4Address) -> bool:
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionIP/ExpanseAggregateAttributionIP.py:17]: Done

File: Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionIP/ExpanseAggregateAttributionIP.py [hunk 15]
```diff
@@ -0,0 +1,106 @@
...
+
+        if sightings is not None:
+            current_state['sightings'] += sightings
+
+    markdown = '## ExpanseAggregateAttributionIP'
+    outputs = list(current_ips.values())
+
+    return CommandResults(
+        readable_output=markdown,
+        outputs=outputs if len(outputs) > 0 else None,
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionIP/ExpanseAggregateAttributionIP.py]: Done

File: Packs/ExpanseV2/Scripts/ExpanseEnrichAttribution/ExpanseEnrichAttribution.py [hunk 16]
```diff
@@ -0,0 +1,101 @@
...
+    return CommandResults(
+        outputs_prefix=outputs_prefix,
+        outputs_key_field=key_field,
+        outputs=current_list if len(current_list) > 0 else None,
+        readable_output=""
+    )
+
+
+def enrich_command(args: Dict[str, Any]) -> CommandResults:
+    type_ = args.pop("type", None)
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseEnrichAttribution/ExpanseEnrichAttribution.py]: Why not? :-) switched to the traditional args.get('type') pattern.

File: Packs/ExpanseV2/Scripts/ExpanseEnrichAttribution/ExpanseEnrichAttribution.py [hunk 17]
```diff
@@ -0,0 +1,101 @@
...
+    current_list = argToList(args.get('current', []))
+    enrich_list = argToList(args.get('enrich', []))
+    enrich_key = args.get('enrich_key', "")
+    enrich_fields = argToList(args.get('enrich_fields', []))
+
+    if type_ == "IP":
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseEnrichAttribution/ExpanseEnrichAttribution.py]: Changed the code to make type argument case insensitive.

File: Packs/ExpanseV2/Scripts/ExpanseEnrichAttribution/ExpanseEnrichAttribution.py [hunk 18]
```diff
@@ -0,0 +1,101 @@
...
+from CommonServerUserPython import *  # noqa
+
+from typing import Dict, Any
+import traceback
+
+
+''' COMMAND FUNCTION '''
+
+
+def enrich(current_list: List[Dict[str, Any]], enrich_list: List[Dict[str, Any]], enrich_key: str,
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseEnrichAttribution/ExpanseEnrichAttribution.py:12]: Done

File: Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py [hunk 19]
```diff
@@ -0,0 +1,121 @@
...
+        md_result.extend(dsToFieldList(next_dict[0], next_dict[1], dict_stack))
+
+    return '\n'.join(md_result)
+
+
+''' COMMAND FUNCTION '''
+
+
+def evidence_dynamic_section(args: Dict[str, Any]) -> CommandResults:
+    incident = demisto.incidents()[0]
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py]: Done

File: Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py [hunk 20]
```diff
@@ -0,0 +1,121 @@
...
+    custom_fields = incident.get('CustomFields')
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py]: Done

File: Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py [hunk 21]
```diff
@@ -0,0 +1,121 @@
...
+        ))
+
+    return md_result
+
+
+def convert_to_markdown(evidence: Dict[str, Any]) -> str:
+    md_result = []
+    dict_stack: List[Tuple[str, Dict[str, Any]]] = []
+
+    dict_stack.append(("Evidence", evidence))
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py]: Done

File: Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py [hunk 22]
```diff
@@ -0,0 +1,121 @@
...
+import json
+import traceback
+from json import JSONDecodeError
+from typing import Dict, Any, Tuple, List, Optional
+
+
+''' STANDALONE FUNCTION '''
+
+
+def valueToMarkdown(v: Any) -> Optional[str]:
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py]: Done

File: Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py [hunk 23]
```diff
@@ -0,0 +1,121 @@
...
+    if isinstance(v, int) or isinstance(v, float) or isinstance(v, str):
+        return f"{v}"
+
+    if isinstance(v, list) and len(v) == 0:
+        return "*empty list*"
+
+    return None
+
+
+def dsToFieldList(name: str, d: Dict[str, Any], dict_stack: List[Tuple[str, Dict[str, Any]]]) -> List[str]:
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseEvidenceDynamicSection/ExpanseEvidenceDynamicSection.py]: Done

File: Packs/ExpanseV2/Scripts/ExpanseRefreshIssueAssets/ExpanseRefreshIssueAssets.py [hunk 24]
```diff
@@ -0,0 +1,72 @@
...
+        asset_key = asset.get('assetkey')
+
+        if asset_type == 'Domain':
+            new_asset = demisto.executeCommand('expanse-get-domain', {"domain": asset_key})
+        elif asset_type == 'IpRange':
+            new_asset = demisto.executeCommand('expanse-get-iprange', {"id": asset_key})
+        elif asset_type == 'Certificate':
+            new_asset = demisto.executeCommand('expanse-get-certificate', {"hash": asset_key})
+        else:
+            # ???
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseRefreshIssueAssets/ExpanseRefreshIssueAssets.py]: :-) switch to a more descriptive comment.

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py [hunk 25]
```diff
@@ -0,0 +1,747 @@
...
+)
+
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()  # pylint: disable=no-member
+
+
+""" CONSTANTS """
+
+SERVER = "https://expander.expanse.co"
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py]: Removed

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py [hunk 26]
```diff
@@ -0,0 +1,747 @@
...
+TOKEN_DURATION = 7200
+PREFIX = SERVER + "/api"
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py]: Removed

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py [hunk 27]
```diff
@@ -0,0 +1,747 @@
...
+        super().__init__(base_url, verify=verify, proxy=proxy, headers=hdr, **kwargs)
+
+    def authenticate(self) -> None:
+        """
+        Perform authentication using API_KEY,
+        stores token and stored timestamp in integration context,
+        retrieves new token when expired
+        """
+        current_utc_timestamp = int(datetime.utcnow().timestamp())
+        token_expiration = current_utc_timestamp + TOKEN_DURATION
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py]: Done

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py [hunk 28]
```diff
@@ -0,0 +1,747 @@
...
+                full_url=next_url,
+                params=params,
+                raise_on_status=True,
+                timeout=30
+            )
+
+            data = result.get('data', [])
+            if data is not None:
+                for a in data:
+                    yield a
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py]: Applied

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py (hunk 4 above)
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py]: Removed

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py (hunk 5 above)
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py]: Applied on all the occurences

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py (hunk 6 above)
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py]: Done (on all of them)

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py [hunk 29]
```diff
@@ -0,0 +1,747 @@
...
+def main() -> None:
+    """main function, parses params and runs command functions
+
+    :return:
+    :rtype:
+    """
+
+    params = demisto.params()
+    api_key = params.get("apikey")
+    base_url = urljoin(params["url"], "/api")
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.py:648]: I don't this is necessary as urljoin itself should take care of the last "/"

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.yml [hunk 30]
```diff
@@ -0,0 +1,160 @@
...
+  required: false
+  type: 8
+- defaultvalue: 'true'
+  display: Fetch indicators
+  name: feed
+  required: false
+  type: 8
+- defaultvalue: '20000'
+  display: The maximum number of indicators to fetch.
+  name: maxIndicators
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.yml]: Done

File: Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.yml [hunk 31]
```diff
@@ -0,0 +1,160 @@
...
+  required: false
+  type: 0
+- defaultvalue: '7'
+  display: Retrieve indicators observed in the last specified number of days
+  name: minLastObserved
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/FeedExpanse.yml]: Not Applied, but renamed to min_last_observed_time

File: Packs/ExpanseV2/Integrations/FeedExpanse/README.md [hunk 32]
```diff
@@ -0,0 +1,66 @@
+Use FeedExpanse to retrieve the list of discovered IPs/Domains/Certificates from Expander
+This integration was integrated and tested with version xx of FeedExpanse
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Integrations/FeedExpanse/README.md]: Done

File: Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.yml (hunk 10 above)
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionDevice/ExpanseAggregateAttributionDevice.yml]: Done, on all of them on all the 3 scripts.

File: Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionIP/ExpanseAggregateAttributionIP.py [hunk 33]
```diff
@@ -0,0 +1,106 @@
...
+
+''' COMMAND FUNCTION '''
+
+
+def aggregate_command(args: Dict[str, Any]) -> CommandResults:
+    input_list = argToList(args.get('input', []))
+    current_list = argToList(args.get('current', []))
+
+    source_ip_fields = argToList(args.get('source_ip_fields', "src,src_ip"))
+    sightings_fields = argToList(args.get('sightings_fields', "count"))
```
[the remark this answers is not in the document]
Comment (jtschichold) [reply Packs/ExpanseV2/Scripts/ExpanseAggregateAttributionIP/ExpanseAggregateAttributionIP.py:74]: Done

File: Packs/ExpanseV2/Integrations/ExpanseV2/ExpanseV2.py [hunk 34]
```diff
@@ -0,0 +1,2330 @@
...
+            url_suffix='/v1/behavior/risky-flows',
+            params=params
+        )
+
+    def parse_asset_data(self, issue: Dict[str, Any],
+                         fetch_details: Optional[bool] = False) -> Tuple[List[Dict[str, Any]], List[str], bool]:
+        assets: List[Dict[str, Any]] = []
+        changed = False
+        ml_feature_list: List[str] = []
+        if 'assets' in issue and isinstance(issue['assets'], list) and len(issue['assets']) > 0:
```
[the remark this answers is not in the document]
Comment (fvigo) [reply Packs/ExpanseV2/Integrations/ExpanseV2/ExpanseV2.py:446]: Changed to:
```
if issue.get('assets') and isinstance(issue['assets'], list):
```
to also make sure it's a list and is iterable

File: Packs/ExpanseV2/Scripts/ExpansePrintSuggestions/ExpansePrintSuggestions.py [hunk 35]
```diff
@@ -0,0 +1,320 @@
...
+        " findings are reported in the following sections.\n"
+    )
+
+    md += "## Service Details\n\n"
+    md += "Logs and asset information were searched for the following service:\n"
+    md += tableToMarkdown(
+        name="Service Information",
+        t=[{"IP": ip, "port": port, "FQDN": fqdn}],
+        headers=["IP", "port", "FQDN"],
+        headerTransform=pascalToSpace,
```
[the remark this answers is not in the document]
Comment (fvigo) [reply Packs/ExpanseV2/Scripts/ExpansePrintSuggestions/ExpansePrintSuggestions.py:149]: fixed

**Comment by fvigo** [comment]
Applied all the changes, only missing README.md for Integration and YML output descriptions, due today.

**Review by Bargenish — APPROVED** [review]

**Pull request merged**